Build failuire against Hyprland 69a4f08dbee6dc08d1e2ce498ce80ab20844e4f3 on NixOS

## Output
```
error: builder for '/nix/store/7y8r4c4h8xra3w02gxap1xljyzpzxc6i-split-monitor-workspaces-0.1.drv' failed with exit code 1;
       last 10 log lines:
       >    63 | struct SConfigValue;
       >       |        ^~~~~~~~~~~~
       > ../src/main.cpp:168:83: error: C99 designator 'intValue' outside aggregate initializer
       >   168 |     HyprlandAPI::addConfigValue(PHANDLE, k_keepFocused, SConfigValue{.intValue = 0});
       >       |                                                                                   ^
       > ../src/main.cpp:168:83: error: invalid use of incomplete type 'struct SConfigValue'
       > /nix/store/2fp6h33gx8iij8iag40lwm455iyzalf0-hyprland-0.35.0+date=2024-02-19_69a4f08-dev/include/hyprland/src/plugins/PluginAPI.hpp:63:8: note: forward declaration of 'struct SConfigValue'
       >    63 | struct SConfigValue;
       >       |        ^~~~~~~~~~~~
       > ninja: build stopped: subcommand failed.
       For full logs, run 'nix log /nix/store/7y8r4c4h8xra3w02gxap1xljyzpzxc6i-split-monitor-workspaces-0.1.drv'.
error: 1 dependencies of derivation '/nix/store/fkhkxaswps0ni4xk1vskjhvjq5j6r0pq-hm_hyprhyprland.conf.drv' failed to build
error: 1 dependencies of derivation '/nix/store/dih7qikmsa2c8w1jwj07vs4bw7bv6mrl-activation-script.drv' failed to build
error: 1 dependencies of derivation '/nix/store/i0yhr5qbzp7y6wjq4a61imhjjb7s38vl-home-manager-generation.drv' failed to build
error: 1 dependencies of derivation '/nix/store/53apwnndyxk1ybhdz60nsl53cr4373ar-unit-home-manager-noah.service.drv' failed to build
error: 1 dependencies of derivation '/nix/store/gxla127dc3ddvag7xs1mxmdpqhb6p8jq-system-units.drv' failed to build
error: 1 dependencies of derivation '/nix/store/4y4whrzj8f4h4a2vslsaa01nx1m8kwc7-etc.drv' failed to build
error: 1 dependencies of derivation '/nix/store/v81hmxwy4x1852lpwj3j6qqidd03sh1x-nixos-system-nixos-24.05.20240216.5863c27.drv' failed to build
```
## Full log
```
@nix { "action": "setPhase", "phase": "unpackPhase" }
Running phase: unpackPhase
unpacking source archive /nix/store/3q7zdqhsm74fczvii07lvaz91h841wal-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: configurePhase
mesonConfigurePhase flags: --prefix=/nix/store/xz69k75al3gy5g1jhk66schxaaps84zg-split-monitor-workspaces-0.1 --libdir=/nix/store/xz69k75al3gy5g1jhk66schxaaps84zg-split-monitor-workspaces-0.1/lib --libexecdir=/>
The Meson build system
Version: 1.3.1
Source dir: /build/source
Build dir: /build/source/build
Build type: native build
Project name: split-monitor-workspaces
Project version: 0.1
C++ compiler for the host machine: g++ (gcc 13.2.0 "g++ (GCC) 13.2.0")
C++ linker for the host machine: g++ ld.bfd 2.40
Host machine cpu family: x86_64
Host machine cpu: x86_64
Compiler for C++ supports arguments -std=c++23: YES
meson.build:8: WARNING: Consider using the built-in option for language standard version instead of using "-std=c++23".
Found pkg-config: YES (/nix/store/knxv5h4hsh86c649rabd6dqfd97kwp5d-pkg-config-wrapper-0.29.2/bin/pkg-config) 0.29.2
Run-time dependency hyprland found: YES 0.35.0
Run-time dependency pixman-1 found: YES 0.43.2
Run-time dependency libdrm found: YES 2.4.120
Build targets in project: 1

split-monitor-workspaces 0.1

  User defined options
    auto_features: enabled
    bindir       : /nix/store/xz69k75al3gy5g1jhk66schxaaps84zg-split-monitor-workspaces-0.1/bin
    buildtype    : plain
    includedir   : /nix/store/xz69k75al3gy5g1jhk66schxaaps84zg-split-monitor-workspaces-0.1/include
    infodir      : /nix/store/xz69k75al3gy5g1jhk66schxaaps84zg-split-monitor-workspaces-0.1/share/info
    libdir       : /nix/store/xz69k75al3gy5g1jhk66schxaaps84zg-split-monitor-workspaces-0.1/lib
    libexecdir   : /nix/store/xz69k75al3gy5g1jhk66schxaaps84zg-split-monitor-workspaces-0.1/libexec
    localedir    : /nix/store/xz69k75al3gy5g1jhk66schxaaps84zg-split-monitor-workspaces-0.1/share/locale
    mandir       : /nix/store/xz69k75al3gy5g1jhk66schxaaps84zg-split-monitor-workspaces-0.1/share/man
    prefix       : /nix/store/xz69k75al3gy5g1jhk66schxaaps84zg-split-monitor-workspaces-0.1
    sbindir      : /nix/store/xz69k75al3gy5g1jhk66schxaaps84zg-split-monitor-workspaces-0.1/sbin
    wrap_mode    : nodownload

Found ninja-1.11.1 at /nix/store/0xns1gyza3sm938pc7klvzrjk5iqam9l-ninja-1.11.1/bin/ninja
mesonConfigurePhase: enabled\ parallel\ building
@nix { "action": "setPhase", "phase": "buildPhase" }
Running phase: buildPhase
build flags: -j16
[1/2] Compiling C++ object libsplit-monitor-workspaces.so.p/src_main.cpp.o
FAILED: libsplit-monitor-workspaces.so.p/src_main.cpp.o
g++ -Ilibsplit-monitor-workspaces.so.p -I. -I.. -I../include -I/nix/store/2fp6h33gx8iij8iag40lwm455iyzalf0-hyprland-0.35.0+date=2024-02-19_69a4f08-dev/include -I/nix/store/2fp6h33gx8iij8iag40lwm455iyzalf0-hypr>
../src/main.cpp: In function 'void mapWorkspacesToMonitors()':
../src/main.cpp:125:41: error: 'class CConfigManager' has no member named 'getConfigValuePtrSafe'; did you mean 'getConfigValuePtr'?
  125 |     int keepFocused = g_pConfigManager->getConfigValuePtrSafe(k_keepFocused)->intValue;
      |                                         ^~~~~~~~~~~~~~~~~~~~~
      |                                         getConfigValuePtr
../src/main.cpp:128:48: error: 'class CConfigManager' has no member named 'getConfigValuePtrSafe'; did you mean 'getConfigValuePtr'?
  128 |         int workspaceCount = g_pConfigManager->getConfigValuePtrSafe(k_workspaceCount)->intValue;
      |                                                ^~~~~~~~~~~~~~~~~~~~~
      |                                                getConfigValuePtr
../src/main.cpp: In function 'PLUGIN_DESCRIPTION_INFO pluginInit(void*)':
../src/main.cpp:167:87: error: C99 designator 'intValue' outside aggregate initializer
  167 |     HyprlandAPI::addConfigValue(PHANDLE, k_workspaceCount, SConfigValue{.intValue = 10});
      |                                                                                       ^
../src/main.cpp:167:87: error: invalid use of incomplete type 'struct SConfigValue'
In file included from /nix/store/2fp6h33gx8iij8iag40lwm455iyzalf0-hyprland-0.35.0+date=2024-02-19_69a4f08-dev/include/hyprland/src/managers/HookSystemManager.hpp:12,
                 from /nix/store/2fp6h33gx8iij8iag40lwm455iyzalf0-hyprland-0.35.0+date=2024-02-19_69a4f08-dev/include/hyprland/src/protocols/Screencopy.hpp:8,
                 from /nix/store/2fp6h33gx8iij8iag40lwm455iyzalf0-hyprland-0.35.0+date=2024-02-19_69a4f08-dev/include/hyprland/src/protocols/ToplevelExport.hpp:6,
                 from /nix/store/2fp6h33gx8iij8iag40lwm455iyzalf0-hyprland-0.35.0+date=2024-02-19_69a4f08-dev/include/hyprland/src/managers/ProtocolManager.hpp:4,
                 from /nix/store/2fp6h33gx8iij8iag40lwm455iyzalf0-hyprland-0.35.0+date=2024-02-19_69a4f08-dev/include/hyprland/src/Compositor.hpp:18,
                 from ../src/main.cpp:3:
/nix/store/2fp6h33gx8iij8iag40lwm455iyzalf0-hyprland-0.35.0+date=2024-02-19_69a4f08-dev/include/hyprland/src/plugins/PluginAPI.hpp:63:8: note: forward declaration of 'struct SConfigValue'
   63 | struct SConfigValue;
      |        ^~~~~~~~~~~~
../src/main.cpp:168:83: error: C99 designator 'intValue' outside aggregate initializer
  168 |     HyprlandAPI::addConfigValue(PHANDLE, k_keepFocused, SConfigValue{.intValue = 0});
      |                                                                                   ^
../src/main.cpp:168:83: error: invalid use of incomplete type 'struct SConfigValue'
/nix/store/2fp6h33gx8iij8iag40lwm455iyzalf0-hyprland-0.35.0+date=2024-02-19_69a4f08-dev/include/hyprland/src/plugins/PluginAPI.hpp:63:8: note: forward declaration of 'struct SConfigValue'
   63 | struct SConfigValue;
      |        ^~~~~~~~~~~~
ninja: build stopped: subcommand failed.
```
## Flake.lock
```
{
  "nodes": {
    "home-manager": {
      "inputs": {
        "nixpkgs": [
          "nixpkgs"
        ]
      },
      "locked": {
        "lastModified": 1708294481,
        "narHash": "sha256-DZtxmeb4OR7iCaKUUuq05ADV2rX8WReZEF7Tq//W0+Y=",
        "owner": "nix-community",
        "repo": "home-manager",
        "rev": "a54e05bc12d88ff2df941d0dc1183cb5235fa438",
        "type": "github"
      },
      "original": {
        "owner": "nix-community",
        "repo": "home-manager",
        "type": "github"
      }
    },
    "hyprland": {
      "inputs": {
        "hyprland-protocols": "hyprland-protocols",
        "hyprlang": "hyprlang",
        "nixpkgs": "nixpkgs",
        "systems": "systems",
        "wlroots": "wlroots",
        "xdph": "xdph"
      },
      "locked": {
        "lastModified": 1708300923,
        "narHash": "sha256-TnZ4E2AZbRXcm5tuior9KMNSjruy11Oqic0yc7ySU1U=",
        "owner": "hyprwm",
        "repo": "Hyprland",
        "rev": "69a4f08dbee6dc08d1e2ce498ce80ab20844e4f3",
        "type": "github"
      },
      "original": {
        "owner": "hyprwm",
        "repo": "Hyprland",
        "type": "github"
      }
    },
    "hyprland-protocols": {
      "inputs": {
        "nixpkgs": [
          "hyprland",
          "nixpkgs"
        ],
        "systems": [
          "hyprland",
          "systems"
        ]
      },
      "locked": {
        "lastModified": 1691753796,
        "narHash": "sha256-zOEwiWoXk3j3+EoF3ySUJmberFewWlagvewDRuWYAso=",
        "owner": "hyprwm",
        "repo": "hyprland-protocols",
        "rev": "0c2ce70625cb30aef199cb388f99e19a61a6ce03",
        "type": "github"
      },
      "original": {
        "owner": "hyprwm",
        "repo": "hyprland-protocols",
        "type": "github"
      }
    },
    "hyprlang": {
      "inputs": {
        "nixpkgs": [
          "hyprland",
          "nixpkgs"
        ]
      },
      "locked": {
        "lastModified": 1708005943,
        "narHash": "sha256-9TT3xk++LI5/SPYgjYX34xZ4ebR93c1uerIq+SE/ues=",
        "owner": "hyprwm",
        "repo": "hyprlang",
        "rev": "aeb3e012adc7b3235335c540b214b82267c2b983",
        "type": "github"
      },
      "original": {
        "owner": "hyprwm",
        "repo": "hyprlang",
        "type": "github"
      }
    },
    "hyprlang_2": {
      "inputs": {
        "nixpkgs": [
          "hyprland",
          "xdph",
          "nixpkgs"
        ]
      },
      "locked": {
        "lastModified": 1704287638,
        "narHash": "sha256-TuRXJGwtK440AXQNl5eiqmQqY4LZ/9+z/R7xC0ie3iA=",
        "owner": "hyprwm",
        "repo": "hyprlang",
        "rev": "6624f2bb66d4d27975766e81f77174adbe58ec97",
        "type": "github"
      },
      "original": {
        "owner": "hyprwm",
        "repo": "hyprlang",
        "type": "github"
      }
    },
    "nix-filter": {
      "locked": {
        "lastModified": 1693833173,
        "narHash": "sha256-hlMABKrGbEiJD5dwUSfnw1CQ3bG7KKwDV+Nx3bEZd7U=",
        "owner": "numtide",
        "repo": "nix-filter",
        "rev": "ac030bd9ba98e318e1f4c4328d60766ade8ebe8b",
        "type": "github"
      },
      "original": {
        "owner": "numtide",
        "repo": "nix-filter",
        "type": "github"
      }
    },
    "nixpkgs": {
      "locked": {
        "lastModified": 1707546158,
        "narHash": "sha256-nYYJTpzfPMDxI8mzhQsYjIUX+grorqjKEU9Np6Xwy/0=",
        "owner": "NixOS",
        "repo": "nixpkgs",
        "rev": "d934204a0f8d9198e1e4515dd6fec76a139c87f0",
        "type": "github"
      },
      "original": {
        "owner": "NixOS",
        "ref": "nixos-unstable",
        "repo": "nixpkgs",
        "type": "github"
      }
    },
    "nixpkgs_2": {
      "locked": {
        "lastModified": 1708118438,
        "narHash": "sha256-kk9/0nuVgA220FcqH/D2xaN6uGyHp/zoxPNUmPCMmEE=",
        "owner": "NixOS",
        "repo": "nixpkgs",
        "rev": "5863c27340ba4de8f83e7e3c023b9599c3cb3c80",
        "type": "github"
      },
      "original": {
        "owner": "NixOS",
        "ref": "nixos-unstable",
        "repo": "nixpkgs",
        "type": "github"
      }
    },
    "nur": {
      "locked": {
        "lastModified": 1708322991,
        "narHash": "sha256-faAddXI8xOjs3u5TcMvGLkcWVjaFVWJjG9G3lsvIVwU=",
        "owner": "nix-community",
        "repo": "NUR",
        "rev": "119ba100d25078d371915db9b8598a5e68cd874f",
        "type": "github"
      },
      "original": {
        "owner": "nix-community",
        "repo": "NUR",
        "type": "github"
      }
    },
    "root": {
      "inputs": {
        "home-manager": "home-manager",
        "hyprland": "hyprland",
        "nixpkgs": "nixpkgs_2",
        "nur": "nur",
        "split-monitor-workspaces": "split-monitor-workspaces"
      }
    },
    "split-monitor-workspaces": {
      "inputs": {
        "hyprland": [
          "hyprland"
        ],
        "nix-filter": "nix-filter"
      },
      "locked": {
        "lastModified": 1698179193,
        "narHash": "sha256-zDW+72enuRGuLLRbO1+pYk8FgRjQrBGksIAOk88OwOI=",
        "owner": "Duckonaut",
        "repo": "split-monitor-workspaces",
        "rev": "2b1abdbf9e9de9ee660540167c8f51903fa3d959",
        "type": "github"
      },
      "original": {
        "owner": "Duckonaut",
        "repo": "split-monitor-workspaces",
        "type": "github"
      }
    },
    "systems": {
      "locked": {
        "lastModified": 1689347949,
        "narHash": "sha256-12tWmuL2zgBgZkdoB6qXZsgJEH9LR3oUgpaQq2RbI80=",
        "owner": "nix-systems",
        "repo": "default-linux",
        "rev": "31732fcf5e8fea42e59c2488ad31a0e651500f68",
        "type": "github"
      },
      "original": {
        "owner": "nix-systems",
        "repo": "default-linux",
        "type": "github"
      }
    },
    "wlroots": {
      "flake": false,
      "locked": {
        "host": "gitlab.freedesktop.org",
        "lastModified": 1706359063,
        "narHash": "sha256-5HUTG0p+nCJv3cn73AmFHRZdfRV5AD5N43g8xAePSKM=",
        "owner": "wlroots",
        "repo": "wlroots",
        "rev": "00b869c1a96f300a8f25da95d624524895e0ddf2",
        "type": "gitlab"
      },
      "original": {
        "host": "gitlab.freedesktop.org",
        "owner": "wlroots",
        "repo": "wlroots",
        "rev": "00b869c1a96f300a8f25da95d624524895e0ddf2",
        "type": "gitlab"
      }
    },
    "xdph": {
      "inputs": {
        "hyprland-protocols": [
          "hyprland",
          "hyprland-protocols"
        ],
        "hyprlang": "hyprlang_2",
        "nixpkgs": [
          "hyprland",
          "nixpkgs"
        ],
        "systems": [
          "hyprland",
          "systems"
        ]
      },
      "locked": {
        "lastModified": 1706521509,
        "narHash": "sha256-AInZ50acOJ3wzUwGzNr1TmxGTMx+8j6oSTzz4E7Vbp8=",
        "owner": "hyprwm",
        "repo": "xdg-desktop-portal-hyprland",
        "rev": "c06fd88b3da492b8f9067be021b9184f7012b5a8",
        "type": "github"
      },
      "original": {
        "owner": "hyprwm",
        "repo": "xdg-desktop-portal-hyprland",
        "type": "github"
      }
    }
  },
  "root": "root",
  "version": 7
}
```

--- a/src/main.cpp
+++ b/src/main.cpp
@@ -122,10 +122,13 @@ void mapWorkspacesToMonitors()
 
     int workspaceIndex = 1;
 
-    int keepFocused = g_pConfigManager->getConfigValuePtrSafe(k_keepFocused)->intValue;
+    static auto* const workspaceCountPtr = (Hyprlang::INT* const*)HyprlandAPI::getConfigValue(PHANDLE, k_workspaceCount)->getDataStaticPtr();
+    static auto* const keepFocusedPtr = (Hyprlang::INT* const*)HyprlandAPI::getConfigValue(PHANDLE, k_keepFocused)->getDataStaticPtr();
+
+    int keepFocused = **keepFocusedPtr;
+    int workspaceCount = **workspaceCountPtr;
 
     for (auto& monitor : g_pCompositor->m_vMonitors) {
-        int workspaceCount = g_pConfigManager->getConfigValuePtrSafe(k_workspaceCount)->intValue;
         std::string logMessage =
             "[split-monitor-workspaces] Mapping workspaces " + std::to_string(workspaceIndex) + "-" + std::to_string(workspaceIndex + workspaceCount - 1) + " to monitor " + monitor->szName;
 
@@ -164,8 +167,8 @@ APICALL EXPORT PLUGIN_DESCRIPTION_INFO PLUGIN_INIT(HANDLE handle)
 {
     PHANDLE = handle;
 
-    HyprlandAPI::addConfigValue(PHANDLE, k_workspaceCount, SConfigValue{.intValue = 10});
-    HyprlandAPI::addConfigValue(PHANDLE, k_keepFocused, SConfigValue{.intValue = 0});
+    HyprlandAPI::addConfigValue(PHANDLE, k_workspaceCount, Hyprlang::INT{10});
+    HyprlandAPI::addConfigValue(PHANDLE, k_keepFocused, Hyprlang::INT{0});
 
     HyprlandAPI::addDispatcher(PHANDLE, "split-workspace", splitWorkspace);
     HyprlandAPI::addDispatcher(PHANDLE, "split-movetoworkspace", splitMoveToWorkspace);

